### PR TITLE
fix: add fpm_model.json to the package tarball before uploading to the registry

### PR DESCRIPTION
**Pull request by henilp105**
This PR introduces a bug fix to attach a `fpm_model.json` file , this bug has been patched in an interim manner in the registry repo: https://github.com/fortran-lang/registry/pull/75 . 

Thanks and Regards,
Henil

### CODE REVIEW

**Comment by gnikit** [comment]
@henilp105 does this affect any other fpm activity? As in, do we call `git_archive` in any a context other than the registry integration at the moment?

**Comment by henilp105** [comment]
@gnikit No, The `git_archive` functionality is only used to publish a tarball to the registry. This bug doesn't affect any other fpm functionality. the tarball uploaded on the registry must contain a `fpm_model.json` file, which was not getting attached in `v0.10.1` to handle this in an interim manner a patch https://github.com/fortran-lang/registry/pull/75 has been made on the registry side to generate and attach the `fpm_model.json` file to  the tarball, This is a permanent fix for the bug.

File: src/fpm/git.f90 [hunk 1]
```diff
@@ -424,20 +424,22 @@ pure function descriptor_name(descriptor) result(name)
     end function descriptor_name
 
   !> Archive a folder using `git archive`.
-  subroutine git_archive(source, destination, ref, verbose, error)
+  subroutine git_archive(source, destination, ref, additional_files, verbose, error)
```
Comment (perazz) [inline src/fpm/git.f90:427]: I believe `git archive` is used as a `zip` wrapper such that no further dependencies need be added to the fpm project. So, we can't put `fpm_model.json` here

**Comment by perazz** [comment]
My edit reflects @gnikit (and my) concern to not lose the option to call `git_archive` from elsewhere. LGTM with this edit, let's see what Giannis thinks.

File: src/fpm/git.f90 (hunk 1 above)
Comment (henilp105) [reply src/fpm/git.f90:427]: sure, Thanks @perazz .

**Comment by henilp105** [comment]
Thanks @perazz for fixing this.

**Review by perazz — APPROVED** [review]

**Review by gnikit — APPROVED** [review]

**Pull request merged**